ourhexenv.py

```diff
@@ -151,30 +154,43 @@ def __init__(self, board_size=5, sparse_flag=True):
         self.left_virtual = self.top_virtual + 2          # player_2 owns left + right nodes
         self.right_virtual = self.top_virtual + 3
 
-    def reset(self):
+    def reset(self, seed: int = None, options: dict = {}):
         self.board = np.zeros((self.board_size, self.board_size), dtype=int)
+        self.agents = self.possible_agents[:]
         
         self.is_first = True
         self.is_pie_rule_usable = True
         self.is_pie_rule_used = False
         self.agent_selection = "player_1"
```

Comment:
Remove this line as it is overridden by using the agent selector a couple of lines below.